### PR TITLE
T-23 husky

### PR DESCRIPTION
**Pull request by mozolevsky**
1. Added changes to esLint config
2. Added pre-push hook (before to be added in github files will be checked using eslint, prettier and jest)
3. Added functionality for checking only staged files

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -4,5 +4,7 @@
       "javascriptreact",
       { "language": "typescript", "autoFix": true },
       { "language": "typescriptreact", "autoFix": true }
-    ]
+    ],
+    "eslint.alwaysShowStatus": true,
```
Comment (DariaYarohovich) [inline .vscode/settings.json]: Where is the status displayed?

File: src/App/App.tsx [hunk 2]
```diff
@@ -1,9 +1,10 @@
 import * as React from 'react';
+import {BrowserRouter as Router, Switch, Route} from 'react-router-dom';
```
Comment (DariaYarohovich) [inline src/App/App.tsx]: I suggest to refuse renaming imports. It makes it more difficult to read and comprehend the code.

File: src/App/MainRoutes.tsx [hunk 3]
```diff
@@ -1,17 +1,13 @@
...
-            <Router>
-                <section>
-                    <Route path="/" exact={true} component={Main} />
-                    <Route path="/finance" exact={true} component={Finance} />
-                </section>
-            </Router>
-        );
-    }
-}
+export const MainRoutes = () => (
```
Comment (DariaYarohovich) [inline src/App/MainRoutes.tsx]: I guess it worth wrapping the component in the `React.memo`.

File: src/Modules/Finance/index.ts [hunk 4]
```diff
@@ -1 +1 @@
-export {FinanceLayout as Finance} from './FinanceLayout';
```
Comment (DariaYarohovich) [inline src/Modules/Finance/index.ts]: Let's not rename exports. Please, leave a TODO.

File: src/Modules/Finance/FinanceLayout.tsx [hunk 5]
```diff
@@ -3,13 +3,9 @@ import {Header} from './Header/Header';
...
-    public render(): React.ReactNode {
-        return (
-            <div className={styles.FinanceLayout}>
-                <h1>Finance module here</h1>
-                <Header />
-            </div>
-        );
-    }
-}
+export const FinanceLayout = () => (
```
Comment (Pavel-Yarakhovich) [inline src/Modules/Finance/FinanceLayout.tsx]: Should we add some TS stuff to this line?

File: src/Modules/Finance/Header/Header.tsx [hunk 6]
```diff
@@ -26,24 +25,29 @@ export class Header extends React.Component<IHeaderProps, IHeaderState> {
...
                 <TargetSum
-                    value={this.state.targetSumValue}
-                    buttonValue={
-                        this.state.targetSumConfirmed ? 'Correct' : 'Confirm'
-                    }
+                    value={targetSumValue}
+                    buttonValue={targetSumConfirmed ? 'Correct' : 'Confirm'}
                     clicked={this.targetSumConfirmedHandler}
                     changed={this.targetSumInputChangedHandler}
-                    confirmed={this.state.targetSumConfirmed}
```
Comment (Pavel-Yarakhovich) [inline src/Modules/Finance/Header/Header.tsx]: Just for me to understand: should I comment things like unneccessary 'this.state' in this line or it's too obvious to be commented?

File: .eslintrc.js [hunk 7]
```diff
@@ -1,18 +1,10 @@
 module.exports = {
     env: {
-        browser: true,
-        es6: true
+		browser: true,
+		node: true,
```
[the remark this answers is not in the document]
Comment (mozolevsky) [reply .eslintrc.js]: An environment defines global variables that are predefined.
node - Node.js global variables and Node.js scoping.
jest - Jest global variables.

Refs: 
https://eslint.org/docs/user-guide/configuring
https://nodejs.org/api/globals.html
https://jestjs.io/docs/ru/next/api

File: .vscode/settings.json (hunk 1 above)
Comment (mozolevsky) [reply .vscode/settings.json]: The status is shown in VS Code status bar

File: src/App/App.tsx (hunk 2 above)
Comment (mozolevsky) [reply src/App/App.tsx]: fixed

File: src/App/MainRoutes.tsx (hunk 3 above)
Comment (mozolevsky) [reply src/App/MainRoutes.tsx]: added

File: src/Modules/Finance/index.ts (hunk 4 above)
Comment (mozolevsky) [reply src/Modules/Finance/index.ts]: fixed

File: src/Modules/Finance/FinanceLayout.tsx (hunk 5 above)
Comment (mozolevsky) [reply src/Modules/Finance/FinanceLayout.tsx]: I would say in the future we will  do this, I mean when some props will be specified.

File: src/Modules/Finance/Header/Header.tsx (hunk 6 above)
Comment (mozolevsky) [reply src/Modules/Finance/Header/Header.tsx]: From my point you should comment as much as you can. But plz explane more broader what is wrong with this piece of code?

**Pull request merged**